File: C++/Lab_11/main.cpp
/**
 * Program Name: main.cpp
 * Developer: Derek P Sifford
 * Date: 01.11.2016
 * Description: Program that demonstrates the extClockType derived class
 */
#include <iostream>
#include <cassert>
#include "extClockType.h"
using namespace std;

int main() {
    int h, m, s;
    extClockType c;

    assert(c.getTimeZone() == "UTC");

    c.getTime(h, m, s);
    assert(h == 0 && m == 0 && s == 0);

    c.setTime(5, 30, 12);
    c.getTime(h, m, s);
    assert(h == 5 && m == 30 && s == 12);

    extClockType c2(5, 10, 15, "EST");
    c2.getTime(h, m, s);
    assert(c2.getTimeZone() == "EST");
    assert(h == 5 && m == 10 && s == 15);

    cout << "==== Tests Passed ====" << endl;

    return 0;
}

File: Rust/lab_11/src/main.rs
// Program that demonstrates the ClockTZ "derived class" for Clock "class"

mod clock_type;
mod ext_clock_type;
use ext_clock_type::*;

fn main() {
    let mut c = ClockTZ::new(5, 30, 12, "UTC");

    let (hr, min, sec) = c.clock.get_time();
    assert_eq!(hr, 5);
    assert_eq!(min, 30);
    assert_eq!(sec, 12);
    assert_eq!(c.get_timezone(), "UTC");

    c.clock.set_time(9, 42, 33);
    assert_eq!(c.clock.get_hours(), 9);
    assert_eq!(c.clock.get_minutes(), 42);
    assert_eq!(c.clock.get_seconds(), 33);

    c.set_timezone("EST");
    assert_eq!(c.get_timezone(), "EST");
}
